src/hotspot/share/opto/vectornode.hpp

```diff
@@ -1686,4 +1686,19 @@ class CountTrailingZerosVNode : public VectorNode {
   virtual int Opcode() const;
 };
 
+class ReverseVNode : public VectorNode {
+public:
+  ReverseVNode(Node* in, const TypeVect* vt)
+  : VectorNode(in, vt) {}
+
+  virtual int Opcode() const;
+};
+
+class ReverseBytesVNode : public VectorNode {
+public:
+  ReverseBytesVNode(Node* in, const TypeVect* vt)
+  : VectorNode(in, vt) {}
+
+  virtual int Opcode() const;
+};
```

Comment:
Just a little tip: Can we add Identify for BYTE type, that is ReverseBytesV(x) ->x.

Reply:
Thanks @theRealELiu ,  its handled on the lines of Revers transformation during final graph reshaping.

Reply:
FTR, Reverse[bytes] related transforms are special since they cannot be done locally in associated Identity routines since corresponding IR  nodes may be succeeded by blends which are generated after creation of Reverse[bytes] IR, thus GVN related routines for both blend and Reverse[bytes] IR may not be able to handle this transform,  strict pattern match based transform during final graph re-shaping looks promising here.

Reply:
Hi @jatin-bhateja ，currently I'm looking at https://bugs.openjdk.org/browse/JDK-8290485, and noticed that you said the transformations for Reverse[bytes] are special that we cannot do the transformations when it is used by a VectorBlend. Could you please explain why? Per my understanding, the masked Reverse[bytes] with VectorBlend should also work, no matter  what changes are applied to the Reverse[bytes], since we only need to make sure the non-masked lanes  come from the first input, and the masked lanes from the second (Reverse[bytes]).  Please incorrect me if I misunderstood. Thanks a lot!

Reply:
Do you mean the `Flag_is_predicated_using_blend` used in the Idea is used to prevent optimizations like `ReverseBytesV (ReverseBytesV X , MASK) , MASK =>  X` when `ReverseBytesV` is predicated but implemented with `VectorBlend` ? So if it is,  then I think the input of ReverseBytesV should be `VectorBlend X (ReverseBytesV X) MASK` while not `ReverseBytesV X`, right?  Thanks!

Reply:
Hi @XiaohongGong , May be I was not explicit, currently we are skipping any ideal transformation for operations on predicated operations implemented using blend and  flag is set prior to a GVN transform, thus it can be safely used in ReverseBytes ideal routine. Since earlier these was no such flag and Blend node was created after GVN transform so it was not possible to skip transform for ReverseBytes[2] node.

![image](https://user-images.githubusercontent.com/59989778/179898611-1f3f061a-f9c5-49e1-8469-8e802c4ce245.png)

Reply:
Yeah, I know that is used to skip the transformations. I'm just curious why we need to skip the optimizations if it is used by a `VectorBlend`. Did you met any issues or something others? Thanks! I want to add a transformation such as `(ReverseBytesV src) => src` if the basic element type is T_BYTE. I cannot see the reason that I need to also skip this transformation for VectorBlend used ReverseBytesV. Maybe we can continue the discussion after I created a PR for it. WDYT?

Reply:
Please refer to the PR https://github.com/openjdk/jdk/pull/9565. Thanks a lot!

Reply:
> Yeah, I know that is used to skip the transformations. I'm just curious why we need to skip the optimizations if it is used by a `VectorBlend`. Did you met any issues or something others? Thanks! I want to add a transformation such as `(ReverseBytesV src) => src` if the basic element type is T_BYTE. I cannot see the reason that I need to also skip this transformation for VectorBlend used ReverseBytesV. Maybe we can continue the discussion after I created a PR for it. WDYT?

It was purely to skip matching over bigger graph pallet at that point, with blends pattern matching will be slightly complex and will need to be handled in VectorBlend idealization routines.

Reply:
I think we can have a separate PR to handle cascaded complementary masked operations over non-predicated targets to handle all the cases in one shot i.e. REVERSE, REVERSEBYTES, NOT.

Other interesting transforms could be  ROR X+ ROL X by same shift = X,   
complementary casts/re-interprets.,  X2Y + Y2X = X